### PR TITLE
fixed maintenance_exclusion deletion

### PR DESCRIPTION
**Pull request by edwardmedia**
Fixes https://github.com/hashicorp/terraform-provider-google/issues/8566

If this PR is for Terraform, I acknowledge that I have:

- [x] Searched through the [issue tracker](https://github.com/hashicorp/terraform-provider-google/issues) for an open issue that this either resolves or contributes to, commented on it to claim it, and written "fixes {url}" or "part of {url}" in this PR description. If there were no relevant open issues, I opened one and commented that I would like to work on it (not necessary for very small changes).
- [ ] [Generated Terraform](https://github.com/GoogleCloudPlatform/magic-modules#generating-downstream-tools), and ran `make test` and `make lint` to ensure it passes unit and linter tests.
- [x] Ensured that all new fields I added that can be set by a user appear in at least one [example](https://github.com/GoogleCloudPlatform/magic-modules/tree/master/templates/terraform/examples) (for generated resources) or [third_party test](https://github.com/GoogleCloudPlatform/magic-modules/tree/master/third_party/terraform/tests) (for handwritten resources or update tests).
- [x] [Ran](https://github.com/hashicorp/terraform-provider-google/blob/master/.github/CONTRIBUTING.md#tests) relevant acceptance tests (If the acceptance tests do not yet pass or you are unable to run them, please let your reviewer know).
- [x] Read the [Release Notes Guide](https://github.com/GoogleCloudPlatform/magic-modules/blob/master/.ci/RELEASE_NOTES_GUIDE.md) before writing my release note below.

**Release Note Template for Downstream PRs (will be copied)**

```release-note:bug
container: Fixed failure in deleting `maintenance_exclusion` for `google_container_cluster`
```

### CODE REVIEW

File: mmv1/third_party/terraform/tests/resource_container_cluster_test.go.erb [hunk 1]
```diff
@@ -3775,14 +3775,9 @@ resource "google_container_cluster" "with_maintenance_exclusion_window" {
 		start_time = "%s"
 		end_time = "%s"
 	}
-	maintenance_exclusion {
```
Comment (rileykarson) [inline mmv1/third_party/terraform/tests/resource_container_cluster_test.go.erb:3778]: Why remove the second? A cluster can have multiple active exclusions.

File: mmv1/third_party/terraform/tests/resource_container_cluster_test.go.erb [hunk 2]
```diff
@@ -1449,7 +1449,7 @@ func TestAccContainerCluster_withMaintenanceExclusionWindow(t *testing.T) {
 				ImportStateVerify:   true,
 			},
 			{
-				Config: testAccContainerCluster_withExclusion_DailyMaintenanceWindow(cluster, "2020-01-01T00:00:00Z", "2020-01-02T00:00:00Z", "2020-05-01T00:00:00Z", "2020-05-02T00:00:00Z"),
```
Comment (rileykarson) [inline mmv1/third_party/terraform/tests/resource_container_cluster_test.go.erb:1452]: Can you add an update case? Ideally we'd go from 2 -> 1.

File: mmv1/third_party/terraform/tests/resource_container_cluster_test.go.erb (hunk 1 above)
Comment (edwardmedia) [reply mmv1/third_party/terraform/tests/resource_container_cluster_test.go.erb:3778]: That test has two steps. The first step contains two `maintenance_exclusion` already and the 2nd step is for `Update`(2->1)  (as you suggested below) while other updating features remain

File: mmv1/third_party/terraform/tests/resource_container_cluster_test.go.erb (hunk 2 above)
Comment (edwardmedia) [reply mmv1/third_party/terraform/tests/resource_container_cluster_test.go.erb:1452]: Explained above

File: mmv1/third_party/terraform/tests/resource_container_cluster_test.go.erb (hunk 1 above)
Comment (rileykarson) [reply mmv1/third_party/terraform/tests/resource_container_cluster_test.go.erb:3778]: Thanks, yep! Sorry I'd misread this.

**Review by rileykarson — APPROVED** [review]

**Pull request merged**